### ISSUE TITLE
Problem with update manager notifier and its preferences

### ISSUE THREAD

**Issue by tomas-52** [linuxmint/mint20.2-beta#23]
I am running the beta version from a live USB stick but do not think the problem could be there.
On boot the notifier icon does not appear on the panel.
The preferences changed already after some update in version 20.1, I think the first refresh was set to 10 minutes.
I like to apply the updates before starting my work, usually I am not in such a hurry to put it off.
So I have set the first refresh to 0 to know if there are updates or not as soon as possible but usually it takes about 3 minutes, that seems too long to me.
In beta I have tested 0 and 1 minute for first refresh, always re-logging for the test, but even after 3 minutes the icon does not appear on the panel.
It gets there only after running update manager from Administration menu.
(no notification is displayed, 3 updates are reported in the manager window).

**Comment by clefebvre** [comment]
Hi Thomas,

In Startup Applications there's an entry for mintupdate and in there you can see there's a delay of 20 seconds in place. This is done to make sure your session has time to load properly and your internet connection to establish itself.

Also in mintupdate, check the logs to make sure that first update is successful.

**Comment by tomas-52** [comment]
I forgot to mention that I am running the Mate version, sorry for that.
Thank you for the information, I rechecked the behavior according to that and made these observations:
Having set the clock to show seconds, first refresh to 0 and recurrent refresh to 1 minute and re-logging got the following times
6:38 login and wired connection info
6:40 no icon in tray, manual start of update manager, immediately found 3 updates
6:41 (according to view info in manager) recurrent refresh in 1 minute
6:42 update manager in tray mode, 17 updates found, further refresh in 1 minute intervals.

So I think there is no change to my previous observation - there is no icon in tray and no notification for update manager even if there are updates available and the time for first refresh has long expired. (and I would add that the setting for first refresh is not obvious as there are different values in Startup Applications and the Update manager itself).

**Comment by tomas-52** [comment]
I have checked the behavior of the update manager when I close the application running in tray.
It seems to stop the manager.
After about 3 minutes wait and manually restarting the update manager it displayed just the following info, and nothing more:
06.22@07:45 ++ Launching Update Manager
06.22@07:45 ++ Changes to the package cache detected, triggering refresh
06.22@07:45 ++ Starting refresh (local only)
06.22@07:45 ++ Found 3 software updates
06.22@07:45 ++ Refresh finished
06.22@07:46 ++ Recurring refresh will happen in 0 day(s), 0 hour(s) and 1 minute(s)
06.22@07:47 ++ Update Manager is in tray mode, performing recurring refresh
06.22@07:47 ++ Starting refresh (retrieving lists of updates from remote servers)
06.22@07:47 ++ Found 17 software updates
06.22@07:47 ++ Refresh finished
06.22@07:47 ++ Recurring refresh will happen in 0 day(s), 0 hour(s) and 1 minute(s)

**Comment by tomas-52** [comment]
I came back to the problem to verify if it has something to do with the launcher in Startup Applications. Yes it has.
I do not understand python scripts, but seeing that both the update launcher and the welcome launcher are both python scripts and they do not start the app I tried to see what is there. According to my little knowledge of programming they will not run on live media.
So I had to install the system to a free partition on my HD.
The update manager icon now appears in tray after login, but still there is no notification.
(I have set the preferences for notifications to show for all updates - only show ... to NO - and both show a notification to 2 days, did not update yet so the last value should not matter.)
In the list of updates there is Intel-microcode version 3.20210608... so I think it is definitely more than 2 days old and a notification should be displayed.

Just by the way, in the beginning I just wanted to do the testing from the live media, but having to install it I came to another problem - the installer (or os-prober) did not find OpenSUSE Leap distribution installed to a BTRFS formated partition on my HD.

**Comment by clefebvre** [comment]
OK, I can see a series of issues but it's not an easy fix. It's down to the way mintupdate works...

It has a cache watcher mechanism which is triggered almost instantly at the beginning and which issues a reload without APT update.

It has a security mechanism which ignores values of 0 and defaults to 1 minute minimum.

It uses the same code path for initial and loop refresh.

When an update is ongoing it ignores any other one, so if the cache watcher updates without an APT reload, it bypasses the initial refresh if it happens to early.

Last but not least if flatpak automation is on, it runs that before launching the actual manager, so that delays it as well.

Basically as it is it's not designed to satisfy this use case. We can work on this as a feature request for 20.3 and hopefully redesign the way this works but there's potential for regression here and we'll need to be careful.

I'm moving this issue to mintupdate.
